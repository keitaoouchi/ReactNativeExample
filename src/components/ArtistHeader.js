import React from "react";
import {Animated, Easing, Text, View} from "react-native";
import ListenOnSpotifyButton from "./ListenOnSpotifyButton";

export default class ArtistHeader extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      fadeAnim: new Animated.Value(0),
    };
  }

  render() {
    return (
      <View style={ style.root }>
        {this.props.artist.imageURL ? <Animated.Image
          source={{uri: this.props.artist.imageURL}}
          style={ [style.image, {opacity: this.state.fadeAnim}]}
          onLoadEnd={() => {
            Animated.timing(
              this.state.fadeAnim,
              {
                toValue: 1,
                easing: Easing.easeInOut,
              },
            ).start();
          }}
        />
          : null
        }
        <Text style={style.artistName}>{this.props.artist.name}</Text>
        <ListenOnSpotifyButton url={this.props.artist.spotifyURL}/>
      </View>
    );
  }
}

const style = {
  root: {
    width: 320,
    height: 300,
    justifyContent: 'flex-end',
    alignItems: 'center',
    alignSelf: 'center',
  },
  image: {width: 160, height: 160, borderRadius: 80},
  artistName: {fontSize: 20, margin: 16, color: 'white'},
};