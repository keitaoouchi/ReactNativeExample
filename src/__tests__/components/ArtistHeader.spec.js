import "react-native";
import React from "react";
import renderer from "react-test-renderer";
import ArtistHeader from "../../../src/components/ArtistHeader";
import ArtistFactory from "../factories/ArtistFactory";

describe('ArtistHeader', () => {

  it('renders correctly', () => {
    const artist = ArtistFactory.build('Artist');
    const tree = renderer.create(
      <ArtistHeader artist={artist}/>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
