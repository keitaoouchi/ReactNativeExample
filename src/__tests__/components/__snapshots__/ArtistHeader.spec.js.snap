// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistHeader renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "center",
      "height": 300,
      "justifyContent": "flex-end",
      "width": 320,
    }
  }
>
  <Image
    onLoadEnd={[Function]}
    source={
      Object {
        "uri": "https://example.com/1",
      }
    }
    style={
      Object {
        "borderRadius": 80,
        "height": 160,
        "opacity": 0,
        "width": 160,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 20,
        "margin": 16,
      }
    }
  >
    Name-1
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#1ed760",
            "borderRadius": 22,
            "height": 44,
            "margin": 12,
            "padding": 8,
            "width": 200,
          },
          false,
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "bold",
            "margin": 4,
            "textAlign": "center",
          }
        }
      >
        Spotifyで聴く
      </Text>
    </View>
  </View>
</View>
`;
